metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbTcpSender.java

```diff
@@ -0,0 +1,99 @@
+import com.izettle.metrics.influxdb.data.InfluxDbWriteObject;
+import com.izettle.metrics.influxdb.utils.InfluxDbWriteObjectSerializer;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An implementation of InfluxDbSender that uses TCP Connection.
+ */
+public class InfluxDbTcpSender implements InfluxDbSender {
+
+    private static final Charset UTF_8 = StandardCharsets.UTF_8;
+
+    private final InfluxDbWriteObject influxDbWriteObject;
+    private final InfluxDbWriteObjectSerializer influxDbWriteObjectSerializer;
+    private final String hostname;
+    private final int port;
+    private final int socketTimeout;
+    private Socket tcpSocket;
+
+    /**
+     * Creates an instance of [[InfluxDbTcpSender]]
+     *
+     * @param hostname          The hostname to connect
+     * @param port              The port to connect
+     * @param database          The database to write into
+     * @param timePrecision     The time precision to use
+     * @param socketTimeout     A socket timeout to use
+     */
+    public InfluxDbTcpSender(final String hostname, final int port, final String database, final TimeUnit timePrecision,
+                             final int socketTimeout) {
+        this.hostname = hostname;
+        this.port = port;
+        this.socketTimeout = socketTimeout;
+
+        this.influxDbWriteObject = new InfluxDbWriteObject(database, timePrecision);
+        this.influxDbWriteObjectSerializer = new InfluxDbWriteObjectSerializer();
+    }
+
+    @Override
+    public void flush() {
+        influxDbWriteObject.setPoints(new HashSet<InfluxDbPoint>());
+    }
+
+    @Override
+    public boolean hasSeriesData() {
+        return influxDbWriteObject.getPoints() != null && !influxDbWriteObject.getPoints().isEmpty();
+    }
+
+    @Override
+    public void appendPoints(InfluxDbPoint point) {
+        if (point != null) {
+            influxDbWriteObject.getPoints().add(point);
+        }
+    }
+
+    @Override
+    public int writeData() throws Exception {
+        retryConnect();
+
+        final String line = influxDbWriteObjectSerializer.getLineProtocolString(influxDbWriteObject);
+
+        OutputStream outputStream = tcpSocket.getOutputStream();
+        outputStream.write(line.getBytes(UTF_8));
+        outputStream.flush();
+
+        return 0;
+    }
+
+    @Override
+    public void setTags(Map<String, String> tags) {
+        if (tags != null) {
+            influxDbWriteObject.setTags(tags);
+        }
+    }
+
+    @Override
+    public Map<String, String> getTags() {
+        return influxDbWriteObject.getTags();
+    }
+
+    private void retryConnect() throws IOException {
+        if (tcpSocket == null) {
+            this.tcpSocket = new Socket(hostname, port);
```

Comment:
There is a findbugs error when using an insecure TCP port
```
[INFO] Unencrypted socket to com.izettle.metrics.influxdb.InfluxDbTcpSender (instead of SSLSocket) [com.izettle.metrics.influxdb.InfluxDbTcpSender] At InfluxDbTcpSender.java:[line 93] UNENCRYPTED_SOCKET
```
But I think this can be added into exclude file. And users have to aware that they are on insecure connections.